### PR TITLE
Update `conda` recipes for Enhanced Compatibility effort

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR updates the `conda` recipe build strings and `cudatoolkit` version specifications as part of the Enhanced Compatibility efforts.

The build strings have been updated to only include the major CUDA version (i.e. `librmm-21.12.00a-cuda11_gc781527_12.tar.bz2`) and the `cudatoolkit` version specifications will now be formatted like `cudatoolkit >=x,<y.0a0` (i.e. `cudatoolkit >=11,<12.0a0`).

Moving forward, we'll build the packages with a single CUDA version (i.e. `11.4`) and test them in environments with different CUDA versions (i.e. `11.0`, `11.2`, `11.4`, etc.).

### CODE REVIEW

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
rerun tests

**Review by raydouglass — APPROVED** [review]

**Review by kkraus14 — DISMISSED** [review]
Please share the results of the discussion with regards to handling of PTX before merging

**Comment by robertmaynard** [comment]
> Please share the results of the discussion with regards to handling of PTX before merging

From the C++ side we either emit SASS for all the necessary architectures so they are not concerned about PTX issues, or we have projects like jitify convert to SASS at runtime and therefore don't run into this problem.

In some cases, when using a new runtime with an older driver, Numba will currently emit PTX not understood by the driver. We are working with the Numba team to provide a fix for this ( via something like ptxcompiler_static ) before the 21.12 release finalizes. In the interim, users of nightlies will be need to run on a driver new enough to run the code emitted by their runtime or temporarily downgrade to an older runtime.

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Comment by kkraus14** [comment]
No worries on dismissing the review. I wonder if at a certain point we should just stop shipping PTX in the libcudf binaries because one of three things will happen if the driver needs to interact with the libcudf PTX:

1) The driver errors from getting a newer PTX version than it supports due to us needing to be aggressive in adopting new CUDA Toolkit versions.
2) We hit a newer than currently supported architecture and the driver tries to JIT compile the PTX into SASS and it presents itself as a process hang to the user because of the size of libcudf.
3) We hit an older unsupported architecture and the driver tries to JIT compile the PTX into SASS and it presents itself as a process hang to the user because of the size of libcudf. At some point it errors either during this JIT compilation or during runtime from the use of architecture features not present on the older architecture.

All of these feel like a distinctly worse user experience than just not shipping the PTX and having the driver error from failing to find SASS for the current device architecture.

**Comment by JohnZed** [comment]
@kkraus14 #2 is an interesting point in particular.... That's basically the specific usecase that PTX is meant to support, but I agree that the JIT times can be very surprising. I think this is the norm for most shipping CUDA software, though, since when new architectures come out, it's also very unpleasant if there is no ability for them to use popular OSS CUDA code at all. I think the ideal would be if we could detect the need to JIT for a new version and output a clear message saying "first time JIT, this will take a while." I don't know of a way to intercept that JIT and issue a message though.

**Comment by JohnZed** [comment]
Asked around, and it sounds like there is not currently an API to intercept when JIT'ing happens and share a message, unfortunately.

**Pull request merged**
